# Project 1 MCS 472 

Abraham Morales Arroyo 

## The Winters in Greenbay, Wisconsin

January 2023 appears to be one for the warmest winter months, at least in Greenbay, Wisconsin. 

Download data for the first winter month, starting December 21 to January 20, for the past 100 years. 

Look for minima, maxima, and average temperatures for each day.

Analyze the data with plots and segmentation, that is: divide the 100 years up in decades. 


#### Disclamer: The project has been modified to take place in Greenbay, Wisconsin. The data has been modified from 1922- 1932 then there is a time skip up to 2012- 2022.  

I am going to have two different files, where the first file containes the Minimum Average, Max Average, and Average Data for December and January, respectfully. This file will be caled ``Classtoday.txt``, and it is from the year 1922- 1932. Then there is another file called ``Classtoday2.txt`` Where it has the same data, but instead it is from the years 2012-2022. 


In [1]:
using DelimitedFiles

In [2]:
A = readdlm("Classtoday.txt")

23×7 Matrix{Any}:
     "Year"    "AMiD"    "AMAD"    "AD"    "AMiJ"    "AMaJ"    "AJ"
 1922        21.18     31.81     26.5     5.65     22.85     14.25
 1923        22.3      32.9      27.64   11.2      25.65     18.43
 1924        22.34     32.9      27.6    -3.25     25.65      5.65
 1925        -0.36     14.18      6.91    9.9      23.65     16.78
 1926        12.91     27.91     20.41   14.95     28.95     21.95
 1927        14.0      27.82     20.91    7.75     22.6      15.18
 1928        18.36     30.55     24.46   17.3      30.9      24.1
 1929        16.91     31.55     24.23   -2.95     13.95      5.5
 1930        13.73     28.0      20.86    9.0      22.3      15.65
 1931        31.818    38.46     35.14   17.8      31.05     24.43
 1932        25.18     37.09     31.14   22.5      33.0      27.75
 2012        13.0      26.36     19.68   14.05     32.45     23.25
 2013         4.09     21.09     12.59   11.7      30.65     21.175
 2014        21.545    31.727    26.64    2.

# The Headers here are abreviations: 

 AMiD= Average Minimum Temperature in December 

 AMAD= Average Maximum Temperature in December

 AD= Average December Temperature

 AMiJ= Average Minimum Temperature in January 

 AMAJ= Average Maximum Temperature in January

AJ= Average January Temperature


For the previous instruction to work, the file ``Classtoday.txt`` has to be on file.

To make the notebook self contained, the string representation of the file will be copied and pasted into a cell.

In [3]:
strA = string(A)

"Any[\"Year\" \"AMiD\" \"AMAD\" \"AD\" \"AMiJ\" \"AMaJ\" \"AJ\"; 1922 21.18 31.81 26.5 5.65 22.85 14.25; 1923 22.3 32.9 27.64 11.2 25.65 18.43; 1924 22.34 32.9 27.6 -3.25 25.65 5.65; 1925 -0.36 14.18 6.91 9.9 23.65 16.78; 1926 12.91 27.91 20.41 14.95 28.95 21.95; 1927 14.0 27.82 20.91 " ⋯ 387 bytes ⋯ ".82 35.73 29.27 8.6 23.3 15.95; 2017 -3.28 13.36 5.05 12.95 25.4 19.18; 2018 22.27 34.09 28.68 7.4 23.95 15.68; 2019 28.18 39.09 33.64 8.6 23.3 15.95; 2020 15.64 31.55 23.59 17.75 32.75 25.25; 2021 17.36 33.36 25.36 20.7 30.9 25.8; 2022 9.27 26.09 17.68 6.8 24.15 15.48]"

In [4]:
print(strA)

Any["Year" "AMiD" "AMAD" "AD" "AMiJ" "AMaJ" "AJ"; 1922 21.18 31.81 26.5 5.65 22.85 14.25; 1923 22.3 32.9 27.64 11.2 25.65 18.43; 1924 22.34 32.9 27.6 -3.25 25.65 5.65; 1925 -0.36 14.18 6.91 9.9 23.65 16.78; 1926 12.91 27.91 20.41 14.95 28.95 21.95; 1927 14.0 27.82 20.91 7.75 22.6 15.18; 1928 18.36 30.55 24.46 17.3 30.9 24.1; 1929 16.91 31.55 24.23 -2.95 13.95 5.5; 1930 13.73 28.0 20.86 9.0 22.3 15.65; 1931 31.818 38.46 35.14 17.8 31.05 24.43; 1932 25.18 37.09 31.14 22.5 33.0 27.75; 2012 13.0 26.36 19.68 14.05 32.45 23.25; 2013 4.09 21.09 12.59 11.7 30.65 21.175; 2014 21.545 31.727 26.64 2.45 20.75 11.6; 2015 25.73 34.82 30.27 7.5 21.45 14.48; 2016 22.82 35.73 29.27 8.6 23.3 15.95; 2017 -3.28 13.36 5.05 12.95 25.4 19.18; 2018 22.27 34.09 28.68 7.4 23.95 15.68; 2019 28.18 39.09 33.64 8.6 23.3 15.95; 2020 15.64 31.55 23.59 17.75 32.75 25.25; 2021 17.36 33.36 25.36 20.7 30.9 25.8; 2022 9.27 26.09 17.68 6.8 24.15 15.48]

In [5]:
B=Any["Year" "AMiD" "AMAD" "AD" "AMiJ" "AMaJ" "AJ"; 1922 21.18 31.81 26.5 5.65 22.85 14.25; 1923 22.3 32.9 27.64 11.2 25.65 18.43; 1924 22.34 32.9 27.6 -3.25 25.65 5.65; 1925 -0.36 14.18 6.91 9.9 23.65 16.78; 1926 12.91 27.91 20.41 14.95 28.95 21.95; 1927 14.0 27.82 20.91 7.75 22.6 15.18; 1928 18.36 30.55 24.46 17.3 30.9 24.1; 1929 16.91 31.55 24.23 -2.95 13.95 5.5; 1930 13.73 28.0 20.86 9.0 22.3 15.65; 1931 31.818 38.46 35.14 17.8 31.05 24.43; 1932 25.18 37.09 31.14 22.5 33.0 27.75; 2012 13.0 26.36 19.68 14.05 32.45 23.25; 2013 4.09 21.09 12.59 11.7 30.65 21.175; 2014 21.545 31.727 26.64 2.45 20.75 11.6; 2015 25.73 34.82 30.27 7.5 21.45 14.48; 2016 22.82 35.73 29.27 8.6 23.3 15.95; 2017 -3.28 13.36 5.05 12.95 25.4 19.18; 2018 22.27 34.09 28.68 7.4 23.95 15.68; 2019 28.18 39.09 33.64 8.6 23.3 15.95; 2020 15.64 31.55 23.59 17.75 32.75 25.25; 2021 17.36 33.36 25.36 20.7 30.9 25.8; 2022 9.27 26.09 17.68 6.8 24.15 15.48]

23×7 Matrix{Any}:
     "Year"    "AMiD"    "AMAD"    "AD"    "AMiJ"    "AMaJ"    "AJ"
 1922        21.18     31.81     26.5     5.65     22.85     14.25
 1923        22.3      32.9      27.64   11.2      25.65     18.43
 1924        22.34     32.9      27.6    -3.25     25.65      5.65
 1925        -0.36     14.18      6.91    9.9      23.65     16.78
 1926        12.91     27.91     20.41   14.95     28.95     21.95
 1927        14.0      27.82     20.91    7.75     22.6      15.18
 1928        18.36     30.55     24.46   17.3      30.9      24.1
 1929        16.91     31.55     24.23   -2.95     13.95      5.5
 1930        13.73     28.0      20.86    9.0      22.3      15.65
 1931        31.818    38.46     35.14   17.8      31.05     24.43
 1932        25.18     37.09     31.14   22.5      33.0      27.75
 2012        13.0      26.36     19.68   14.05     32.45     23.25
 2013         4.09     21.09     12.59   11.7      30.65     21.175
 2014        21.545    31.727    26.64    2.

The instructions in the notebook will continue to work with ``B`` so everything will work even if the file was not present.


# Now we are going to have the rest of the data, From the years 2012-2022, be read and then stored into a matrix so it can be used for this project: 

In [6]:
C = readdlm("Classtoday2.txt")

12×7 Matrix{Any}:
     "Year"    "AMiD"    "AMAD"    "AD"    "AMiJ"    "AMaJ"    "AJ"
 2012        13.0      26.36     19.68   14.05     32.45     23.25
 2013         4.09     21.09     12.59   11.7      30.65     21.175
 2014        21.545    31.727    26.64    2.45     20.75     11.6
 2015        25.73     34.82     30.27    7.5      21.45     14.48
 2016        22.82     35.73     29.27    8.6      23.3      15.95
 2017        -3.28     13.36      5.05   12.95     25.4      19.18
 2018        22.27     34.09     28.68    7.4      23.95     15.68
 2019        28.18     39.09     33.64    8.6      23.3      15.95
 2020        15.64     31.55     23.59   17.75     32.75     25.25
 2021        17.36     33.36     25.36   20.7      30.9      25.8
 2022         9.27     26.09     17.68    6.8      24.15     15.48

In [7]:
strC = string(C)

"Any[\"Year\" \"AMiD\" \"AMAD\" \"AD\" \"AMiJ\" \"AMaJ\" \"AJ\"; 2012 13.0 26.36 19.68 14.05 32.45 23.25; 2013 4.09 21.09 12.59 11.7 30.65 21.175; 2014 21.545 31.727 26.64 2.45 20.75 11.6; 2015 25.73 34.82 30.27 7.5 21.45 14.48; 2016 22.82 35.73 29.27 8.6 23.3 15.95; 2017 -3.28 13.36 5.05 12.95 25.4 19.18; 2018 22.27 34.09 28.68 7.4 23.95 15.68; 2019 28.18 39.09 33.64 8.6 23.3 15.95; 2020 15.64 31.55 23.59 17.75 32.75 25.25; 2021 17.36 33.36 25.36 20.7 30.9 25.8; 2022 9.27 26.09 17.68 6.8 24.15 15.48]"

In [8]:
D=Any["Year" "AMiD" "AMAD" "AD" "AMiJ" "AMaJ" "AJ"; 2012 13.0 26.36 19.68 14.05 32.45 23.25; 2013 4.09 21.09 12.59 11.7 30.65 21.175; 2014 21.545 31.727 26.64 2.45 20.75 11.6; 2015 25.73 34.82 30.27 7.5 21.45 14.48; 2016 22.82 35.73 29.27 8.6 23.3 15.95; 2017 -3.28 13.36 5.05 12.95 25.4 19.18; 2018 22.27 34.09 28.68 7.4 23.95 15.68; 2019 28.18 39.09 33.64 8.6 23.3 15.95; 2020 15.64 31.55 23.59 17.75 32.75 25.25; 2021 17.36 33.36 25.36 20.7 30.9 25.8; 2022 9.27 26.09 17.68 6.8 24.15 15.48]

12×7 Matrix{Any}:
     "Year"    "AMiD"    "AMAD"    "AD"    "AMiJ"    "AMaJ"    "AJ"
 2012        13.0      26.36     19.68   14.05     32.45     23.25
 2013         4.09     21.09     12.59   11.7      30.65     21.175
 2014        21.545    31.727    26.64    2.45     20.75     11.6
 2015        25.73     34.82     30.27    7.5      21.45     14.48
 2016        22.82     35.73     29.27    8.6      23.3      15.95
 2017        -3.28     13.36      5.05   12.95     25.4      19.18
 2018        22.27     34.09     28.68    7.4      23.95     15.68
 2019        28.18     39.09     33.64    8.6      23.3      15.95
 2020        15.64     31.55     23.59   17.75     32.75     25.25
 2021        17.36     33.36     25.36   20.7      30.9      25.8
 2022         9.27     26.09     17.68    6.8      24.15     15.48

# Let us take the average of minimum temperatures of December for the first decade, 1922- 1932

In [9]:
using Statistics

here we are making the selection of the month that we want 

In [10]:
B[1,2]

"AMiD"

In [11]:
mean(B[2:12,2]) # column 2, rows 3 and 4 just as an example

18.033454545454546

 As we can see the average temperature for December 20th - 31st from 1922-1932 is about ``18.03`` (rounded) 

# Now we are going to be comparing this to the minimum temperature of Dec 20th- 31st from 2012-2022: 

Making the selection of the month: 

In [12]:
D[1,2]

"AMiD"

now we are going to be taking the average: 

In [13]:
mean(D[2:12,2]) # Remember that for the years 2012-2022 is going to be stored in the matrix named ``D``

16.056818181818183

Here we see that we got the average of minimum average temperature for December, 2012-2022, to be ``16.06``, rounded

# Outcome: 

When we compare the average temperature from 1922-1932, Average Min December: ``18.03``. To the data 2012-2022: ``16.06`` we see that for the years 2012-2022 the average temperature is much lower than the average temperature from 1922-1932.  

There is a difference of: 

In [14]:
18.03-16.06

1.9700000000000024

The difference from the years 1922-1932 to 2012-2022 is: ``1.9700000000000024``, meaning it has become colder.

# We will now do the same but for the average Max temperature for December 1922-1932

In [15]:
B[1,3]

"AMAD"

This is the Average Max temperature for December 1922-1932. Now we are going to make sure that we are working with the correct rolls and data:

In [16]:
B[2:12,3]# this gives us the data

11-element Vector{Any}:
 31.81
 32.9
 32.9
 14.18
 27.91
 27.82
 30.55
 31.55
 28.0
 38.46
 37.09

Yes this matches up the data for Average Max temperature for December of 1922-1932.

In [17]:
mean(B[2:12,3]) # column 2, rows 3 and 4 just as an example

30.288181818181815

This is the average for Average Max temperature for December 1922-1932, ``30.29``

# Now we are going to be comparing this to the Max Average temperature of Dec 20th- 31st from 2012-2022: 

In [18]:
D[1,3]

"AMAD"

In [19]:
D[2:12,3]# this gives the data

11-element Vector{Any}:
 26.36
 21.09
 31.727
 34.82
 35.73
 13.36
 34.09
 39.09
 31.55
 33.36
 26.09

In [20]:
mean(D[2:12,3]) # column 2, rows 3 and 4 just as an example

29.751545454545454

The average of ``29.75`` is for Average Max December 2012-2022

# Outcome: 

Here the 1922- 1932 is : ``30.29``

For 2012-2022 is: ``29.75``


Here we see that the temperature has dropped, from ``30.29`` to ``29.75``. The difference is: 


In [21]:
30.29-29.75

0.5399999999999991

Here we see that the difference was ``0.54``, meaning it has become colder.

# Now we are going to be comparing this to the average temperature of Dec 20th- 31st from 1922-1932: 

In [22]:
B[1,4]

"AD"

In [23]:
B[2:12,4]# this gives the data

11-element Vector{Any}:
 26.5
 27.64
 27.6
  6.91
 20.41
 20.91
 24.46
 24.23
 20.86
 35.14
 31.14

In [24]:
mean(B[2:12,4])

24.16363636363636

Here we see that the average of temperature is ``24.16`` for December 1922-1932.

# Now we are going to be comparing this to the average temperature of Dec 20th- 31st from 2012-2022: 

In [25]:
C[1,4]

"AD"

In [26]:
C[2:12,4]# this gives the data

11-element Vector{Any}:
 19.68
 12.59
 26.64
 30.27
 29.27
  5.05
 28.68
 33.64
 23.59
 25.36
 17.68

In [27]:
mean(C[2:12,4])

22.95

Here we see that the average temperature for December 2012-2022 is: ``22.95``

# Outcome: 

Average December Temperature 1922-1932: ``24.16``

Average December Temperature 2012-2012: ``22.95``


Here we see that the average temperature from one decade to another decade has dropped. It went from ``24.16`` to ``22.95``

In [28]:
24.16-22.95

1.2100000000000009

Here we see that there is a decrease of ``1.21`` in temperature. 

# Now we are going to find the Average Minimum temperature for January from 1922-1932: 

In [32]:
B[1,5] #this gives the max minimum to January

"AMiJ"

In [33]:
B[2:12,5]# this gives the data

11-element Vector{Any}:
  5.65
 11.2
 -3.25
  9.9
 14.95
  7.75
 17.3
 -2.95
  9.0
 17.8
 22.5

In [35]:
mean(B[2:12,5])

9.986363636363636

Here we see that the average Max for min January 1922-1932 is: ``9.99``

# Now we do the same for January but for 2012-2022:

In [36]:
C[1,5]

"AMiJ"

In [37]:
C[2:12,5]

11-element Vector{Any}:
 14.05
 11.7
  2.45
  7.5
  8.6
 12.95
  7.4
  8.6
 17.75
 20.7
  6.8

In [38]:
mean(C[2:12,5])

10.772727272727273

Here we see that there is ``10.77`` for the Average Jan Minimum for 2012-2022. 

# Outcome: 

Here we see for 1922-1932 for Min Jan average is: 9.99

For 2012- 2022 for the Min Jan Average: 10.77 

Here we see that there is an increase in temperature: 

In [39]:
10.77-9.99

0.7799999999999994

here we see that the increase of ``0.78`` in the temperature. 

# Now we are going to find the Average Max temperature for January from 1922-1932: 

In [40]:
B[1,6] #this gives the max of Jan

"AMaJ"

In [41]:
B[2:12,6]# this gives the data

11-element Vector{Any}:
 22.85
 25.65
 25.65
 23.65
 28.95
 22.6
 30.9
 13.95
 22.3
 31.05
 33.0

In [42]:
mean(B[2:12,6])

25.50454545454546

here we see that the average Max for Jan 1922-1932 is: ``25.50``

# Now we are going to find the Average Max temperature for January from 2012-2022: 

In [43]:
C[1,6]

"AMaJ"

In [44]:
C[2:12,6]

11-element Vector{Any}:
 32.45
 30.65
 20.75
 21.45
 23.3
 25.4
 23.95
 23.3
 32.75
 30.9
 24.15

In [46]:
mean(C[2:12,6])

26.277272727272724

Here we see that the average for Jan 2012-2022 is: ``26.28``

# Outcome 

we see that weather: ``25.50`` (1922-1932)

We see that weather: ``26.28`` (2012-2022)

Here we see that there was increase of weather:

In [47]:
# the difference of weather is: 
26.28-25.50

0.7800000000000011

here we see that there is an increase of weather is: ``0.78``

# Now we are going to find the average of January 1922-1932: 

In [48]:
B[1,7]

"AJ"

In [49]:
B[2:12,7]# this gives the data

11-element Vector{Any}:
 14.25
 18.43
  5.65
 16.78
 21.95
 15.18
 24.1
  5.5
 15.65
 24.43
 27.75

In [50]:
mean(B[2:12,7])

17.242727272727276

Here we see that the average for January temperature for 1922-1932 is: ``17.24``

# Now we do this for January temperature 2012-2022: 

In [51]:
C[1,7]

"AJ"

In [52]:
C[2:12,7]

11-element Vector{Any}:
 23.25
 21.175
 11.6
 14.48
 15.95
 19.18
 15.68
 15.95
 25.25
 25.8
 15.48

In [53]:
mean(C[2:12,7])

18.526818181818182

Here we see that the average for 2012-2022 January: ``18.53`` 

# Outcome 

Temperature for 1922-1932 Jan: ``17.24``

Temperature for 2012-2022 Jan: ``18.53``

There has been an increase in temperature, the difference is: 

In [54]:
18.53-17.24

1.2900000000000027

Here we see that the increase of temperature is ``1.29``

# This below is legacy code: DO NOT MESS WITH IT!!!

In [30]:
D[2:12,2]# this gives the 'first decade'

11-element Vector{Any}:
 13.0
  4.09
 21.545
 25.73
 22.82
 -3.28
 22.27
 28.18
 15.64
 17.36
  9.27

In [45]:
D[2:12,1]# this gives the years 

11-element Vector{Any}:
 2012
 2013
 2014
 2015
 2016
 2017
 2018
 2019
 2020
 2021
 2022